### PR TITLE
Adds the XM51 Breaching Scattergun

### PR DESCRIPTION
**Pull request by VileBeggar**
# About the pull request

[Design doc (a bit out of date).](https://hackmd.io/uVhllc-KQSGcvjhEUw66bw?edit)

The XM51 breaching scattergun is a 16-gauge, pump-action mag-fed shotgun that utilizes specialized breaching shells. It allows marines to break through doors and walls alike with ease at a safe distance. Each shell fires out four groups of pellets, the main one dealing 55 damage and the other three dealing 10 each. Though it has decent damage, this weapon performs weakly in direct combat compared to other options and is entirely designed for a support role. Its maximum effective range is about 4 tiles.

It takes:
- 1 shot for doors
- 2 shots for regular resin walls
- 3 shots for thick resin walls
- 2 shots for regular walls,
- 4 shots for reinforced walls.
- 4 shots for pylons,

Each XM51 kit comes with a dedicated holster for the weapon, along with two spare magazines, two boxes of shells and a stock. Being able to wear the weapon around your waist gives the user a great deal of flexibility when it comes to their loadout. The stock makes it unable to be holstered, but it does give it a very devastating two shot burst-mode that quickly clears out walls.

Ammo for the weapon is a bit scarce but still present in a decent quantity. Requisitions starts with a single 120 round shell box which they can purchase more of, along with a couple of spare magazines. More magazines can be purchased, and ASRS has a chance of sending up spare ammo as well.

This weapon is based on the [actual M51 breaching scattergun.](https://avp.fandom.com/wiki/M51_Breaching_Scattergun)

# Explain why it's good for the game

Marines are very cowardly when it comes to pushing. Especially so when they're facing countless thick resin walls and doors. This contributes to a stale meta where the best option is to just sit behind cades and wait for the other team to make a mistake, because why would you risk dying to take down one resin door? That's not fun for anyone. Marines have a few ways of making some breathing room for themselves from a safe distance, with one of the best ones being the U7 underbarrel shotgun. It's a very good and fun attachment, even putting aside the combat capabilities it has.

However, it's entirely exclusive to the M41A rifles. Any marine that wants to experiment with their weapon choice a bit and forgo the rifle lose one of the best tools they have access to. The XM51 gives marines another, more effective way to breach resin bunkers. In exchange for the ease of use, abundance of ammo and good combat capabilities of the U7, you gain a safer, faster, flexible and more potent tool.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

https://github.com/cmss13-devs/cmss13/assets/17518895/6f4441d2-740e-4110-a9a6-32c39c424576

combat capabilities vs the u7

https://github.com/cmss13-devs/cmss13/assets/17518895/c0cb1359-cfd0-4b86-8c77-3885c86c5c8c

https://github.com/cmss13-devs/cmss13/assets/17518895/834d9bbb-3aee-4203-93a7-af40e28b7805

xm51camo
xm51ammo
xm51statnostock

</details>

# Changelog
:cl:
add: Added the XM51 breaching scattergun, available in small quantities as a restricted weapon within Requisitions.

/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by Zonespace27** [comment]
Willing to try this once sprite approval is gotten

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Review by Birdtalon — COMMENTED** [review]
Cool gun, tested and works as described. Just one suggestion for cleanliness but otherwise looks good to go.

**Review by Birdtalon — APPROVED** [review]
Looks fine and works fine when tested.

**Comment by Zonespace27** [comment]
Issues I've seen so far:
- No inhand sprite
- No equip noise

File: code/game/objects/items/storage/belt.dm
Comment (Zonespace27) [inline code/game/objects/items/storage/belt.dm:972]: are there defines for these?

File: code/modules/projectiles/guns/rifles.dm
Comment (Zonespace27) [inline code/modules/projectiles/guns/rifles.dm:1887]: code doc, and new vars are put below inherited ones.

File: code/modules/projectiles/guns/rifles.dm
Comment (Zonespace27) [inline code/modules/projectiles/guns/rifles.dm:1886]: use the COOLDOWN_XXXX macros

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
Some at-a-glance changes

File: code/game/objects/items/storage/belt.dm
Comment (VileBeggar) [reply code/game/objects/items/storage/belt.dm:972]: afaik all camo swapping instances are handled similar to this, but i could possibly missing something. not sure

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by HaultyAnonie** [comment]
403 forbidden for the design document

**Comment by VileBeggar** [comment]
some balance changes:

- total damage buffed by 60%, 50 -> 80
- can now be worn on marine armor
- accuracy buffed
- max range buffed from 4 to 6
- guncase now contains 2 boxes of shells
- squad preps now have doubled scaling, 2*scale -> 4*scale
- fire sound volume lowered by 2.5db

**Comment by VileBeggar** [comment]
balance change: colony walls now take 2 shells to take down, reinforced walls need 4 shells to take down

**Comment by VileBeggar** [comment]
hopefully the final balance change:

- holster rig can now only hold 2 spare magazines in total instead of 4. instead, you have 5 extra slots that you can store spare shells in.

File: code/game/objects/items/storage/belt.dm
Comment (Drulikar) [inline code/game/objects/items/storage/belt.dm:1244]: You should not be using the _private methods. Override/implement handle_item_insertion and remove_from_storage instead.

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/belt.dm
Comment (Drulikar) [inline code/game/objects/items/storage/belt.dm:1241]: The arguments for remove_from_storage are not these.

See https://github.com/cmss13-devs/cmss13/blob/c1e4a5aae576874607a86f34861a401241f256ff/code/game/objects/items/storage/storage.dm#L526

**Review by Drulikar — DISMISSED** [review]

File: code/game/objects/items/storage/belt.dm
Comment (VileBeggar) [reply code/game/objects/items/storage/belt.dm:1241]: i did not look very far when replacing it, whoops

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by Zonespace27 — DISMISSED** [review]
Lower the range of the gun. As it stands, a regular wall at 5 tiles goes down in 3 shots. Reduce the range to 2-3 tiles. Your PR desc is also not entirely accurate on hits to destroy structures.

**Comment by cuberound** [comment]
please increase range to at least 5, right not it can not hit a wall that is further then 3 tiles from your character, by that i mean your tile, empty empty empty wall, you can not hurth that wall

**Comment by Drulikar** [comment]
Actually turns out the issue is simply that there isn't an image in nano\images\weapons named xm51.png

But its good to have it in the lineart spritesheet. I will probably look at changing this over to the spritesheet - but doesn't need to happen in your PR.

**Review by Drulikar — APPROVED** [review]

**Comment by Drulikar** [comment]
I overlooked this was still lacking sprite approval tag. But whatever needs addressed I'm happy to handle myself need be.

**Pull request merged**
